pkg/reconciler/ingress/controller.go

```diff
@@ -168,3 +189,35 @@ func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl
 
 	return impl
 }
+
+func waitForCache(caches *generator.Caches) {
```

Comment:
Why not drive this by events? Have we considered adding an "observer" pattern to the cache that informs us every time there's a state change to `ingressesToSync`?